src/Defaults/Carrier/Email.php

```diff
@@ -298,4 +304,29 @@ public function allowUnfilteredHtmlBody($carrierData, $rawData)
 
 		return $carrierData;
 	}
+
+	/**
+	 * Retrieve email "From" header from settings.
+	 *
+	 * @return string
+	 */
+	protected function getFromHeaderFromSettings()
+	{
+		$fromHeader = '';
+
+		if (getSetting('carriers/email/from_name')) {
+			$fromHeader = 'From: ' . getSetting('carriers/email/from_name');
+		}
+
+		$fromEmail = getSetting('carriers/email/from_email');
+		if ($fromEmail) {
+			if (strpos($fromHeader, 'From:') === false) {
+				$fromHeader = 'From:';
+			}
+
+			$fromHeader .= ' <' . $fromEmail . '>';
+		}
+
+		return $fromHeader;
```

Comment:
It would be simpler to:
1. Check whether the settings are set in the beginning and early return if not
2. Use `sprintf`
3. Name the function `getFromHeaderSetting`

Reply:
I don't think there is a good way to return early because the two options can be set independently. But I've simplified it to the max because getSetting returns an empty string anyway when it's not set.

But I think these two options should be set as one option, because if we don't set the "carriers/email/from_email" option here, the default WP settings will be set anyway because you can't send an email without an email address in the "from" header.